File: BOJ/[10775] 공항/woong-jae/10775.cpp
#include <iostream>
#include <vector>
#include <cstring>

using namespace std;
int g, p;
vector<int> parent;

int find(int x) {
    if (x == parent[x]) return x;
    else return parent[x] = find(parent[x]);
}

void unite(int x, int y) {
    x = find(x);
    y = find(y);
    parent[x] = y;
}

int main() {
    int cnt = 0;
    cin >> g >> p;
    parent.resize(g + 1);
    for (int i = 1; i <= g; i++) {
        parent[i] = i;
    }
    for (int i = 0; i < p; i++) {
        int gi, gate;
        cin >> gi;
        gate = find(gi);
        if (gate == 0) break;
        unite(gate, gate - 1);
        cnt++;
    }
    cout << cnt << '\n';
}

File: BOJ/[10775] 공항/ONE/10775.py
G = int(input())
P = int(input())
parent = [0]

for i in range(1, G + 1):
    parent.append(i);


def find(n):
    if n == parent[n]:
        return n
    p = find(parent[n])
    parent[n] = p
    return parent[n]


def union(x, y):
    x = find(x)
    y = find(y)
    parent[x] = y


cnt = 0
for _ in range(P):
    g = int(input())
    emptyGate = find(g)

    if emptyGate == 0:
        break
    cnt += 1
    union(emptyGate, emptyGate - 1)

print(cnt)
